kong/runloop/certificate.lua

```diff
@@ -265,6 +265,20 @@ local function execute()
     log(ERR, "could not set configured private key: ", err)
     return ngx.exit(ngx.ERROR)
   end
+
+  if cert_and_key.cert_alt and cert_and_key.key_alt then
+    ok, err = set_cert(cert_and_key.cert_alt)
+    if not ok then
+      log(ERR, "could not set alternate configured certificate: ", err)
+      return ngx.exit(ngx.ERROR)
+    end
+
+    ok, err = set_priv_key(cert_and_key.key_alt)
+    if not ok then
+      log(ERR, "could not set alternate configured private key: ", err)
+      return ngx.exit(ngx.ERROR)
+    end
+  end
```

Comment:
Questions:

What is the certificate selection or preference order when multiple cert-key pairs are present for a single SNI?

This PR adds in an ability to add 2 instead of 1 certificate for an SNI. Should we not make this feature more generic where, for some value of n, n certificates can be associated with a single SNI? Why have an arbitrary limit of 2?


I do get the conundrum of how to avoid the breaking change.
Here is a draft proposal:

Let's change the schema to:
```yaml
certs:
- cert: ""
  key: ""
  bundle: # come up with a better name
  - cert: ""
     key:  ""
```

And then update the validation to ensure that either `.cert` and .key` are provided or validate that len(bundle)>0 and contains valid cert key pairs.

Reply:
It depends on the cipher suite negotiation, the order where set_cert_* is called won't make difference. With [intermediate](https://github.com/Kong/kong/blob/master/kong/conf_loader/init.lua#L30) or old and we let Kong to prefer server ciphers, ECDSA certs will be selected first. In tls1.3 cipher suites doesn't include cert type/signature type but I think most implementations will just select ECDSA over RSA if both present.
That also means there'll only be two kinds of signing algorithm supported by most server/clients. I'd imagine you can set a third eddsa cert but it will never be selected.